### PR TITLE
Obtain raw_seqs remotely if cannot be found locally

### PR DESCRIPTION
**Pull request by IsmailM**
This should fix the problems that @monicadragan was seeing with the get_raw_sequences...

This simply looks online if a sequence cannot be found on the local database...

@yeban, do you mind having a quick look, Thanks

@monicadragan, if you are happy with this, we can close #88...

### CODE REVIEW

**Comment by yeban** [comment]
If I just pass a FASTA file to GV; GV should run BLAST, process output, and do whatever it needs to run the validations. In this scenario, it doesn't make sense if I have to give GV a FASTA file of hit sequences as well - where would I get the FASTA file of hits from? So removing hit sequence retrieval altogether is not an option.

If I additionally tell GV which database to validate against, it's guaranteed that hit sequences can be derived from the same database. In this scenario, it doesn't make sense to fetch hit sequences from NCBI. Furthermore, I may have chosen to validate against sequences that are not available on NCBI.

Finally the scenario, where I choose to run BLAST before hand and pass the XML/TSV output to GV. In this case, either `-d` or `-r` should be enforced. Without `-d` GV can't reliably know which database to extract the hit sequences from. With `-r` GV doesn't have to extract hit sequences itself.

To sum up,

* Fetching hit sequences from NCBI if not found locally is not justified in any of the scenarios. In fact, can lead to errors.
* GV should do as much as possible for the user. So automatically extracting hit sequences should remain. At max you could choose to remove automatic hit sequence extraction for the use case where BLAST output is passed to GV.

**Comment by yeban** [comment]
Regarding the code,

* I don't see why `Sequences` class has any functionality at all to fetch hit sequences. It should perhaps just do something like `RawSequences.fetch(accession)` and happily worth the hit sequence.
* `RawSequence`s should be able to fetch hit sequences either from NCBI or given db, and invoked before `Sequences` gets a chance to call `RawSequence#fetch` - like in https://github.com/wurmlab/genevalidator/commit/b0d5b94aae46e27001c2bb411197df48084ef359.
* I don't think `Sequences` class is appropriately named. `Query` may suit it better?

**Comment by IsmailM** [comment]
@yeban 

I think this is now ready

I have tested the following commands on this:

```
# Fastest option - simply runs validations
$ bundle exec genevalidator -d DB_PATH -x BLAST_XML -r RAW_SEQS INPUT_FILE
$ bundle exec genevalidator -d DB_PATH -t BLAST_TAB -o BLAST_TAB_OUTFMT -r RAW_SEQS INPUT_FILE

# Creates RAW_seqs file automatically
$ bundle exec genevalidator -d DB_PATH -x BLAST_XML INPUT_FILE
$ bundle exec genevalidator -d DB_PATH -t BLAST_TAB -o BLAST_TAB_OUTFMT INPUT_FILE

# Creates BLAST XML and RAW_seqs file automatically
$ bundle exec genevalidator -d DB_PATH INPUT_FILE

$ bundle exec genevalidator INPUT_FILE
# runs BLAST on remote NCBI servers (the default db is 'swissprot -remote)'
```
